ConsigmentConstruct/ConsigmentConstruct/Consigment.cs

```diff
@@ -0,0 +1,117 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ConsigmentConstruct
+{
+    public struct Consigment
+    {
+        public int Quantity;
+        public double Price;
+        
+        public readonly double Cost;
+
+        public Consigment(int quantity, double price)
+        {
+            if (quantity <= 0 || price <=0)
+            {
+                throw new Exception("Вводимые значения должны быть положительными");
+            }
+
+            Quantity = quantity;
+            Price = Math.Round(price,2);
+            Cost = Price*Quantity;
+
+        }
+        public static Consigment Read()
+        {
+            int quantity;
+            double price;
+            
+
+            while (true)
+            {
+                Console.WriteLine("Введите кол-во товара:");
+                quantity = int.Parse(Console.ReadLine());
+
+                Console.WriteLine("Введите цену за 1 товар:");
+                price = Math.Round(double.Parse(Console.ReadLine()),2);
+
+
+
+                if (quantity <= 0 || price <= 0)
+                {
+                    throw new Exception("Вводимые значения должны быть положительными");
+                }
+                else
+                {
+                    break;
+                }
+            }
+
+            return new Consigment(quantity, price);
+        }
+
+        public override string ToString()
+        {
+            return $"{Quantity} шт. по {Price} руб.";
+        }
+
+        public override bool Equals(object obj)
+        {
+            if (obj is Consigment)
+            {
+                Consigment p = (Consigment)obj;
+
+                return Cost * Price == p.Cost * p.Price;
```

Comment:
Неправильно. В задании сказано

> Две партии товара равны, когда их количества и цены на единицу товара совпадают.

То есть
```suggestion
                return Quantity == p.Quantity && Price == p.Price;
```